### DISCUSSION TITLE
Very large differences between cell and pack currents using Equivalent Circuit model

### DISCUSSION THREAD

**Discussion by MustafaDragonborne** [pybamm-team/liionpack#242] [General]
![image](https://user-images.githubusercontent.com/64160706/229102741-e69a9c13-dd98-426e-bf9e-c807d5ba3151.png)

![image](https://user-images.githubusercontent.com/64160706/229102524-afcbd8a6-9cba-4a33-9efe-9ce1f18915e0.png)

I simulated a simple 2p1s configuration using the ECM (not sure if the problem is ECM specific) but I see that the difference between the accumulated cell currents and the total pack current is too large. Sometimes, the pack current is much higher than the accumulated cell currents as is evident from the last graph (Pack current - accumulated cell currents) which means there is alot of extra current coming in. 

This is because of the fact that the cell currents are displaced one step forward which cause the error. This difference becomes negligible if the cell currents are moved one step back with respect to the pack current. This is a bug that needs to be fixed.

![image](https://user-images.githubusercontent.com/64160706/229140958-efca2ff1-3526-46da-a098-015d901875f2.png)

**Comment by TomTranter** [comment]
Thanks for finding this bug. In future please post the steps to reproduce. 
```
import liionpack as lp
import pybamm
import numpy as np
import matplotlib.pyplot as plt

lp.set_logging_level('NOTICE')

# Define parameters
Np = 2
Ns = 1

# Generate the netlist
netlist = lp.setup_circuit(Np=Np, Ns=Ns)

# Define a cycling experiment using PyBaMM
experiment = pybamm.Experiment([
    'Charge at 2 A for 2 minutes',
    'Rest for 2 minutes',
    'Discharge at 1 A for 2 minutes',
    'Rest for 2 minutes'
    ],
    period='10 seconds')

# Define the PyBaMM parameters
parameter_values = pybamm.ParameterValues("Chen2020")

# Solve the pack
output = lp.solve(netlist=netlist,
                  parameter_values=parameter_values,
                  experiment=experiment,
                  initial_soc=0.5,
                  manager='casadi')

sum_cell = np.sum(output['Cell current [A]'], axis=1)
pack = output['Pack current [A]']
diff = sum_cell - pack
if np.any(np.abs(diff)>0.001):
    print('Differences in pack and cell current')
    print('Time steps', np.argwhere(np.abs(diff)>0.001).flatten())
    print(diff[np.abs(diff)>0.001])

plt.figure()
plt.plot(pack)
plt.plot(sum_cell)
```

**Comment by TomTranter** [comment]
I have pushed a fix to the develop branch which is currently working with the PyBaMM develop branch. Both packages will be updated in the next few days on pip

**Comment by TomTranter** [comment]
Please update both PyBaMM and Liionpack versions and see if this fixes the problem

**Reply by MustafaDragonborne** [reply]
Nope now I get this when using the Thevenin Model.
KeyError: "'Open-circuit voltage [V]' not found. Best matches are ['Open circuit voltage [V]', 'Upper voltage cut-off [V]', 'Lower voltage cut-off [V]']"

This error is similar to #237 

EDIT: To solve the error and get it working with the Thevenin model, I had to make 2 changes. I had to change 
1. the "Open circuit voltage [V]" value of my parameter to "Open-circuit voltage [V]"
2. self.variable_names = [
            "Terminal voltage [V]",
            "Surface open-circuit voltage [V]",
        ]
to 
self.variable_names = [
            "Battery voltage [V]",
            "Open-circuit voltage [V]" 
        ]
in line 171 of solvers.py

**Other than these errors, I would say the problem has been solved.**

![image](https://user-images.githubusercontent.com/64160706/230377509-4554a3e5-9d2c-4617-9c96-bd691aea4218.png)

The image captures the error now without the 1 second displacement. Thanks alot @TomTranter

**Reply by Nomados** [reply]
@MustafaDragonborne can you please share the code and versions that you used to get it work?
